client/app/scripts/controllers/create-manual-order.js

```diff
@@ -81,35 +81,41 @@ angular.module('ShoppinPalApp').controller(
 
       this.uploader.onAfterAddingFile = function (fileItem) {
         var filename = fileItem.file.name;
-        var slicedFilename = filename.slice(0,-4);
-        var data = slicedFilename.split('-');
+        var re = /(?:\.([^.]+))?$/;
+        var fileExtension = re.exec(filename);
 
-        if (data[0]===undefined || data[1]===undefined) {
-          self.validUpload = false;
-          $spAlerts.addAlert('Filename is not valid', 'error', 5000);
-        }
-        else {
-          var storeName = data[0];
-          storeName = storeName.replace(/_/g, ' '); // . is treated as regex when
-          //console.log('regex with storeName', storeName);
+        if(fileExtension[1] === 'csv' || fileExtension[1] === 'CSV'){
```

Comment:
You can cover more ground if you use something like: 
1. `if( (fileExtension[1]).toLowerCase === 'csv' ) ) {`, or
2.  `if( (fileExtension[1]).toUpperCase === 'CSV' ) ) {`
because it lets you account for weird things like: `file.cSV` or `file.CsV` etc.

I haven't tested my pseudo-code, just wrote the idea down.